services/src/main/java/org/exoplatform/news/search/NewsIndexingServiceConnector.java

```diff
@@ -0,0 +1,177 @@
+import org.exoplatform.commons.search.index.impl.ElasticIndexingServiceConnector;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.HTMLSanitizer;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.news.NewsService;
+import org.exoplatform.news.model.News;
+import org.exoplatform.services.jcr.ext.app.SessionProviderService;
+import org.exoplatform.services.jcr.ext.common.SessionProvider;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+public class NewsIndexingServiceConnector extends ElasticIndexingServiceConnector {
+  private static final long     serialVersionUID = 1L;
+
+  public static final String    TYPE             = "news";
+
+  private static final Log      LOG              = ExoLogger.getLogger(NewsIndexingServiceConnector.class);
+
+  private final NewsService     newsService;
+
+  private final IdentityManager identityManager;
+
+  public NewsIndexingServiceConnector(IdentityManager identityManager, InitParams initParams, NewsService newsService) {
+    super(initParams);
+    this.newsService = newsService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  @Override
+  public Document create(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public Document update(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public List<String> getAllIds(int offset, int limit) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean canReindex() {
+    return false;
+  }
+
+  private Document getDocument(String id) {
+    if (StringUtils.isBlank(id)) {
+      throw new IllegalArgumentException("id is mandatory");
+    }
+    LOG.debug("Index document for news id={}", id);
+    Document document = new Document();
+    News news = null;
+    String spaceId = null;
+    SessionProvider systemProvider = SessionProvider.createSystemProvider();
+    SessionProviderService sessionProviderService = CommonsUtils.getService(SessionProviderService.class);
+    sessionProviderService.setSessionProvider(null, systemProvider);
+    try {
+      news = newsService.getNewsById(id);
+    } catch (Exception e) {
+      e.printStackTrace();
+    } finally {
+      sessionProviderService.removeSessionProvider(null);
+    }
+    if (news == null) {
+      throw new IllegalStateException("news with id '" + id + "' is mandatory");
+    }
+    Map<String, String> fields = new HashMap<>();
+    fields.put("id", news.getId().replace("comment", ""));
+
+    fields.put("title", news.getTitle());
+
+    String body = news.getBody();
+    if (StringUtils.isBlank(body)) {
+      body = news.getTitle();
```

Comment:
I think that can you can sanitize the news body here instead of doing it after documen t creation